packages/taquito/src/wallet/wallet.ts

```diff
@@ -225,7 +225,12 @@ export class Wallet {
    */
   async pk({ forceRefetch }: PKHOption = {}) {
     if (!this._pk || forceRefetch) {
-      this._pk = await this.walletProvider.getPK();
+      // Try to get PK from wallet provider first, fall back to signer
```

Comment:
After checking the code in `https://github.com/ecadlabs/taquito/blob/bdd33d376640c11a65ee2110de1f728ee8c0e045/packages/taquito/src/taquito.ts#L251-L260` 
A LegacyWalletProvider will be created if no wallet were configured, hence the original line `this._pk = await this.walletProvider.getPK();` in wallet provider is already handling getting pk from signer